glusterd2/commands/devices/transaction.go

```diff
@@ -0,0 +1,29 @@
+package devicecommands
+
+import (
+	"os/exec"
+
+	device "github.com/gluster/glusterd2/glusterd2/device"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+)
+
+func txnPrepareDevice(c transaction.TxnCtx) error {
+	var deviceinfo device.Info
+	if err := c.Get("nodeid", &deviceinfo.NodeID); err != nil {
+		return err
+	}
+	if err := c.Get("devicename", &deviceinfo.DeviceName); err != nil {
+		return err
+	}
+	for _, element := range deviceinfo.DeviceName {
+		pvcreateCmd := exec.Command("pvcreate", "--metadatasize=128M", "--dataalignment=256K", element)
```

Comment:
@atinmu I have used metadatasize and dataalignment as per suggested by rtalur, but I feel it requires your look.